src/neuroglancer/segmentation_user_layer.ts

```diff
@@ -92,9 +93,21 @@ export class SegmentationUserLayerGroupState extends RefCounted implements Segme
   constructor(public layer: SegmentationUserLayer) {
     super();
     const {specificationChanged} = this;
-    this.visibleSegments.changed.add(specificationChanged.dispatch);
     this.hideSegmentZero.changed.add(specificationChanged.dispatch);
     this.segmentQuery.changed.add(specificationChanged.dispatch);
+
+    const {visibleSegments, selectedSegments} = this;
+    visibleSegments.changed.add(specificationChanged.dispatch);
+    selectedSegments.changed.add(specificationChanged.dispatch);
+    selectedSegments.changed.add((x, add) => {
+      if (!add) {
+        if (x) {
+          visibleSegments.delete(x);
+        } else {
+          visibleSegments.clear();
```

Comment:
originally I had this listener adding and deleting to visible segments but we switched to wanting the star button to not make the segment visible. I'm not sure if this just adds more confusion, it means that in some other places I have to specifically add to visible but not when deleting from selected.